app/models/manageiq/providers/cloud_manager.rb

```diff
@@ -159,6 +159,8 @@ def self.display_name(number = 1)
 
     def destroy_mapped_tenants
       if source_tenant
+        # We just destroyed ourself, reload the source_tenant association
+        source_tenant.reload
         source_tenant.all_subtenants.destroy_all
         source_tenant.all_subprojects.destroy_all
         source_tenant.destroy
```

Comment:
Do we want to change this to `destroy!` so errors like this will be caught in the future?

Reply:
I can open a PR to see if that works.  It's a callback so I assume it was written to not raise on purpose but it might just been an oversight.